aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp

```diff
@@ -302,10 +302,18 @@ void ldl_solve_cusolver(
 
 template <typename scalar_t>
 static void apply_triangular_solve(const Tensor& A, const Tensor& B, bool left, bool upper, TransposeType transpose, bool unitriangular) {
+#ifdef ROCM_VERSION
+  // Cannot auto-hipifiy this piece of code, because in other functions the uplo
+  // and other variables need to be hipSOLVER's type.
+  auto uplo = upper ? rocblas_fill_upper : rocblas_fill_lower;
+  const auto trans = (rocblas_operation)to_cublas(transpose);
+  rocblas_side side = left ? rocblas_side_left : rocblas_side_right;
+#else
   cublasFillMode_t uplo = upper ? CUBLAS_FILL_MODE_UPPER : CUBLAS_FILL_MODE_LOWER;
```

Comment:
Does this mean that cusolver and cublas use the same types (at least the ones listed here)?

Reply:
Let's file a JIRA to bring this up with the hipSolver team to see how we can avoid this.